File: api/examples/php/minimal-chat/tester.py
# this is supposed to work but my python wont connect to HTTPS
import json
import requests

# Load the JSON configuration file
with open("tester.json") as file:
    data = json.load(file)

# Send the initial request
response = requests.post(data['url'], headers={"Content-Type": "application/json", "Authorization": "Bearer " + data['token']}, data=json.dumps(data)).json()
if 'choices' in response and response['choices']:
    print("Press CTRL+C to exit...\n" + response['choices'][0]['text'] + "\n> ")
else:
    print("Unable to find response text!")
    exit()

# Interact with the model
try:
    while True:
        user_input = input("> ")
        payload = {
            'session': response['session_id'],
            'messages': [{'role': 'user', 'content': user_input}],
            'passthru': True,
            'stream': True
        }
        response = requests.post(data['url'], headers={"Content-Type": "application/json", "Authorization": "Bearer " + data['token']}, data=json.dumps(payload)).json()
        for message in response['messages']:
            print(message['content'])
except KeyboardInterrupt:
    pass

File: api/examples/php/minimal-chat/tester.php
<?php
$json = json_decode(file_get_contents(__DIR__ . "/tester.json")) or die("Unable to open ./tester.json file!");
$response = json_decode(file_get_contents($json->url, false, stream_context_create(["http" => ["method" => "POST", "header" => "Content-Type: application/json\r\nAuthorization: Bearer " . $json->token . "\r\n", "content" => json_encode($json)]])), true) or die("Unable to decode response!");
echo (isset($response["choices"][0]["message"]["content"]) ? "Press CTRL+C to exit...\n" . $response["choices"][0]["message"]["content"] . "\n> " : die("Unable to find response text!\n"));
while (true) {
    $fp = fopen($json->url, "r", false, stream_context_create(["http" => ["method" => "POST", "header" => "Content-Type: application/json\r\nAuthorization: Bearer " . $json->token . "\r\n", "content" => json_encode(["session" => $response["session_id"], "messages" => [["role" => "user", "content" => trim(fgets(fopen("php://stdin", "r")))]], "passthru" => true, "prompt_tokens" => 512, "stream" => true])]])) or die("Unable to open stream!");
    while (!feof($fp)) echo str_replace("\0", "", fgetc($fp));
    fclose($fp);
    echo ("\n> ");
}
